bin/remigrate_visit_logs.ts

```diff
@@ -0,0 +1,186 @@
+/*
+ * Re-migration of visit logs
+ *
+ * Made necessary because of the problems identified in:
+ * - https://github.com/TwinePlatform/data-sync/issues/39
+ *
+ * Usage:
+ *  npm run exc ./bin/remigrate_visit_logs [source_url] [--ignore=a,b,c] [--ignoreNullAct=a]
+ *
+ * Arguments:
+ *  source_url: Connection URL for PostgreSQL database of original visitor app
+ *
+ * Flags:
+ *  ignore: comma-separated list of qr codes to disregard from the migration
+ *  ignoreNullAct: qr code to disregard if corresponding activity is null (junk log)
+ *
+ * This script assumes only those users already in the target database need their logs
+ * re-migrated. Any errors with migration or merging of user accounts must be handled separately,
+ * before this script is run.
+ *
+ * General algorithm is:
+ * 1. Get all visitors from target database
+ * 2. For each visitor
+ *   2.1 Get list of logs added to target DB (_after_ data sync), A
+ *   2.2 Get list of logs from source DB (_before data sync), B
+ *   2.3 Delete all logs for visitor from target database
+ *   2.4 Insert the logs (A concat B) into the target database
+ */
+import * as parse from 'minimist';
+import * as Knex from 'knex';
+import { Visitors, Organisations } from '../src/models';
+import { getConfig } from '../config';
+
+
+process.on('unhandledRejection', (err) => { throw err; });
+
+const { _: [sourceUrl], ignore, ignoreNullAct } = parse(process.argv.slice(2));
```

Comment:
Does this work? You're using object destructuring on an array...

Reply:
It works, the value of `_` is an array, so I'm using both array destructuring and object destructuring.